### PR TITLE
ApiProxy - implementation change for http requests

### PR DESCRIPTION
**Pull request by zachluna**
1. Change ApiProxy and unit tests from being streamwrapper based to using GuzzleHttp\Client which is curl based. This resolve conflict with APIs that use streamwrappers such as UrlFetch. 
2. Change headers in UrlFetchStream headers to be public to enable reading and verifying successful header responses in tests/code.

### CODE REVIEW

**Review by jinglundong — COMMENTED** [review]
I think this whole VmApiProxy is not used any more (at least the same way as it was intended for). Could you investigate a little bit more about how it's being used today? 

* I'm sharing a shortn link to avoid exposing our base base to the public. http://shortn/_XVRF11VCaz says it's used for service bridge. 
* Service bridge is no longer used by VmRuntimes. See a design I wrote in 2018 :) go/service-bridge-https

**Comment by zachluna** [comment]
> I think this whole VmApiProxy is not used any more (at least the same way as it was intended for). Could you investigate a little bit more about how it's being used today?
> 
> * I'm sharing a shortn link to avoid exposing our base base to the public. http://shortn/_XVRF11VCaz says it's used for service bridge.
> * Service bridge is no longer used by VmRuntimes. See a design I wrote in 2018 :) go/service-bridge-https

I have now converted VmApiProxy to RealApiProxy (and deprecated the prior RealApiProxy). Thanks for pointing this out!

File: src/Runtime/RealApiProxy.php [hunk 1]
```diff
@@ -16,64 +16,252 @@
...
+use google\appengine\ext\remote_api\Request;
+use google\appengine\ext\remote_api\Response;
+use google\appengine\ext\remote_api\RpcError\ErrorCode;
+use Google\AppEngine\Runtime\RPCFailedError;
+use GuzzleHttp\Client;
+
+/**
+ * ApiProxy implementation
+ */
 class RealApiProxy extends ApiProxyBase {
```
Comment (jinglundong) [inline src/Runtime/RealApiProxy.php:28]: As we discussed this afternoon, this is renamed from the VmApiProxy right? Can I treat it as a move/rename and pay less attention to this file?

**Review by jinglundong — COMMENTED** [review]
I didn't review RealApiProxy line by line. I can approve this PR if it's a pure rename, otherwise I need to take a closer look at it. I'm wondering if `git mv` or other techniques could show me a better diff. I want to diff this against the old VmApiProxy.

File: src/Runtime/RealApiProxy.php (hunk 1 above)
Comment (zachluna) [reply src/Runtime/RealApiProxy.php:28]: Yes It's a pure rename, nothing else changed except the name.

**Review by jinglundong — APPROVED** [review]

**Pull request merged**